examples/nextjs-spa/src/pages/registration.tsx

```diff
@@ -15,47 +15,53 @@ import { AxiosError } from "axios"
 // We will use UserAuthCard from Ory Elements to display the registration form.
 import Layout from "@/components/layout"
 import { UserAuthCard } from "@ory/elements"
-import { QueryParams } from "../pkg/helpers"
-import { HandleError } from "../pkg/hooks"
+import { SetUriFlow } from "@/pkg/helpers"
+import { HandleError } from "@/pkg/hooks"
 import { NextPageWithLayout } from "./_app"
 
 const Registration: NextPageWithLayout = () => {
   const [flow, setFlow] = useState<RegistrationFlow>()
 
-  const handleError = HandleError()
-
   // Get flow information from the URL
   const router = useRouter()
 
   const flowId = String(router.query.flow || "")
   const returnTo = String(router.query.return_to || "")
 
-  const getRegistrationFlow = useCallback(
+  const getFlow = useCallback(
     (id: string) =>
       ory
         .getRegistrationFlow({ id })
         .then(({ data }) => {
           // We received the flow - let's use its data and render the form!
           setFlow(data)
+          SetUriFlow(Router, data.id)
         })
-        .catch((error: AxiosError) => handleError(error)),
-    [handleError],
+        .catch(handleError),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I'm not sure if we should disable linter rules like this.
Since the functions are memoized through the `useCallback` function, it should be alright to add them to this dependency list.